test/watch.js

```diff
@@ -0,0 +1,441 @@
+const test = require('ava');
+const fs = require('fs');
+const brunch = require('../lib');
+const {
+  prepareTestDir,
+  teardownTestDir,
+  fileContains,
+  fileDoesNotContains,
+  fileExists,
+  fileDoesNotExists,
+  requestBrunchServer
+} = require('./_test_helper');
+const fixturify = require('fixturify');
+
+var watcher;
+
+const watch = function(...args) {
+  watcher = brunch.watch(...args);
+};
+
+test.beforeEach.cb((t) => {
+  teardownTestDir();
+  prepareTestDir();
+
+  if (watcher) {
+    // close chokidar to prevent that it understands the fixtures being copied as new files being added
+    watcher.watcher.close();
+    if (watcher.server) {
+      watcher.server.close(t.end);
+    } else {
+      t.end();
+    }
+  } else {
+    t.end();
+  }
+});
+
+test.serial.cb('compile on file changes', t => {
+  fixturify.writeSync('.', {
+    'brunch-config.js': `module.exports = {
+      files: {
+        javascripts: {
+          joinTo: 'app.js'
+        }
+      }
+    };`,
+    'app': {
+      'assets': {
+        'index.html': '<h1>hello world</h1>'
+      },
+      'initialize.js': 'console.log("hello world")'
+    }
+  });
+
+  var calls = 0;
+
+  const onCompile = function() {
+    calls++;
+
+    if (calls === 1) {
```

Comment:
Is the conditional needed? I think it can be more linear and straightforward, and less nested, with an assertion with a custom error message instead:

```javascript
t.true(calls === 1, `Unexpected number of calls ${calls}`);
fileExists....
```